src/neptune/internal/backends/hosted_client.py

```diff
@@ -66,6 +68,18 @@
 }
 
 
+class GzipAdapter(HTTPAdapter):
+    def send(self, request, stream=False, **kw):
+        if request.body is not None and not stream:
+            request_body = request.body if isinstance(request.body, bytes) else bytes(request.body, "utf-8")
+            gzip_compress = zlib.compressobj(zlib.Z_DEFAULT_COMPRESSION, zlib.DEFLATED, zlib.MAX_WBITS | 16)
```

Comment:
Can we try catch that wrt. to `https://docs.python.org/3/library/zlib.html#zlib.error`?

Reply:
I don't think it's necessary because an exception is very generic.

Reply:
They mentioned that this could happen on compression as well